Build failed with an exception.

# Bug Report

## Describe the Bug

The build fails with an exception when i run `./gradlew build` from the root directory of MVD.

### Expected Behavior

An completed build process without errors

### Observed Behavior

During the build. I see following output when the error occurs:


```shell
> Task :extensions:superuser-seed:compileJava FAILED
/home/conuser/mvd2/extensions/superuser-seed/src/main/java/org/eclipse/edc/identityhub/seed/ParticipantContextSeedExtension.java:94: error: incompatible types: Map<String,Object> cannot be converted to String
                            .orElse(generatedKey);
                                    ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
1 error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':extensions:superuser-seed:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --info option to get more log output.
> Run with --scan to get full insights.

BUILD FAILED in 27s
48 actionable tasks: 42 executed, 6 up-to-date
```

## Steps to Reproduce

Steps to reproduce the behavior:
1. Pull latest repo `git clone https://github.com/eclipse-edc/MinimumViableDataspace.git mvd2` 
2. Go into the folder `cd mvd2`
3. Run `sudo ./gradlew build`
4. See error

## Context Information

- Cloned Repo from September 17th
- Ubuntu 22.04.4 LTS (GNU/Linux 5.15.0-121-generic x86_64) 

## Detailed Description

Log file with debug parameter.
[gradlew_error.txt](https://github.com/user-attachments/files/17028235/gradlew_error.txt)

## Possible Implementation
None

--- a/deployment/modules/identity-hub/identityhub.tf
+++ b/deployment/modules/identity-hub/identityhub.tf
@@ -57,7 +57,7 @@ resource "kubernetes_deployment" "identityhub" {
           }
           port {
             container_port = var.ports.ih-identity-api
-            name           = "did-mgmt"
+            name           = "identity"
           }
           port {
             container_port = var.ports.ih-did

--- a/extensions/superuser-seed/src/main/java/org/eclipse/edc/identityhub/seed/ParticipantContextSeedExtension.java
+++ b/extensions/superuser-seed/src/main/java/org/eclipse/edc/identityhub/seed/ParticipantContextSeedExtension.java
@@ -91,7 +91,7 @@ public class ParticipantContextSeedExtension implements ServiceExtension {
                                         .onFailure(f -> monitor.warning("Error overriding API key for '%s': %s".formatted(superUserParticipantId, f.getFailureDetail())));
                                 return key;
                             })
-                            .orElse(generatedKey);
+                            .orElse(generatedKey.get("apiKey").toString());
                     monitor.info("Created user 'super-user'. Please take note of the API Key: %s".formatted(apiKey));
                 })
                 .orElseThrow(f -> new EdcException("Error creating Super-User: " + f.getFailureDetail()));

--- a/extensions/superuser-seed/src/test/java/org/eclipse/edc/identityhub/seed/ParticipantContextSeedExtensionTest.java
+++ b/extensions/superuser-seed/src/test/java/org/eclipse/edc/identityhub/seed/ParticipantContextSeedExtensionTest.java
@@ -27,6 +27,8 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 
+import java.util.Map;
+
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.contains;
@@ -41,6 +43,7 @@ import static org.mockito.Mockito.when;
 class ParticipantContextSeedExtensionTest {
 
     public static final String SUPER_USER = "super-user";
+    public static final String API_KEY = "apiKey";
     private final ParticipantContextService participantContextService = mock();
     private final Vault vault = mock();
     private final Monitor monitor = mock();
@@ -55,9 +58,9 @@ class ParticipantContextSeedExtensionTest {
 
     @Test
     void start_verifySuperUser(ParticipantContextSeedExtension ext,
-                                    ServiceExtensionContext context) {
+                               ServiceExtensionContext context) {
 
-        when(participantContextService.createParticipantContext(any())).thenReturn(ServiceResult.success("some-key"));
+        when(participantContextService.createParticipantContext(any())).thenReturn(ServiceResult.success(Map.of(API_KEY, "some-key")));
         ext.initialize(context);
 
         ext.start();
@@ -81,7 +84,7 @@ class ParticipantContextSeedExtensionTest {
 
     @Test
     void start_withApiKeyOverride(ParticipantContextSeedExtension ext,
-                                       ServiceExtensionContext context) {
+                                  ServiceExtensionContext context) {
 
 
         when(vault.storeSecret(any(), any())).thenReturn(Result.success());
@@ -91,7 +94,7 @@ class ParticipantContextSeedExtensionTest {
                 .thenReturn(apiKeyOverride);
 
         when(participantContextService.createParticipantContext(any()))
-                .thenReturn(ServiceResult.success("generated-api-key"));
+                .thenReturn(ServiceResult.success(Map.of(API_KEY, "generated-api-key")));
         when(participantContextService.getParticipantContext(eq(SUPER_USER)))
                 .thenReturn(ServiceResult.notFound("foobar"))
                 .thenReturn(ServiceResult.success(superUserContext().build()));
@@ -106,7 +109,7 @@ class ParticipantContextSeedExtensionTest {
 
     @Test
     void start_withInvalidKeyOverride(ParticipantContextSeedExtension ext,
-                                           ServiceExtensionContext context) {
+                                      ServiceExtensionContext context) {
         when(vault.storeSecret(any(), any())).thenReturn(Result.success());
 
         var apiKeyOverride = "some-invalid-key";
@@ -114,7 +117,7 @@ class ParticipantContextSeedExtensionTest {
                 .thenReturn(apiKeyOverride);
 
         when(participantContextService.createParticipantContext(any()))
-                .thenReturn(ServiceResult.success("generated-api-key"));
+                .thenReturn(ServiceResult.success(Map.of(API_KEY, "generated-api-key")));
         when(participantContextService.getParticipantContext(eq(SUPER_USER)))
                 .thenReturn(ServiceResult.notFound("foobar"))
                 .thenReturn(ServiceResult.success(superUserContext().build()));
@@ -130,7 +133,7 @@ class ParticipantContextSeedExtensionTest {
 
     @Test
     void start_whenVaultReturnsFailure(ParticipantContextSeedExtension ext,
-                                            ServiceExtensionContext context) {
+                                       ServiceExtensionContext context) {
         when(vault.storeSecret(any(), any())).thenReturn(Result.failure("test-failure"));
 
         var apiKeyOverride = "c3VwZXItdXNlcgo=.asdfl;jkasdfl;kasdf";
@@ -138,7 +141,7 @@ class ParticipantContextSeedExtensionTest {
                 .thenReturn(apiKeyOverride);
 
         when(participantContextService.createParticipantContext(any()))
-                .thenReturn(ServiceResult.success("generated-api-key"));
+                .thenReturn(ServiceResult.success(Map.of(API_KEY, "generated-api-key")));
         when(participantContextService.getParticipantContext(eq(SUPER_USER)))
                 .thenReturn(ServiceResult.notFound("foobar"))
                 .thenReturn(ServiceResult.success(superUserContext().build()));

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -117,11 +117,12 @@ edc-sql-ih-didstore-sql = { module = "org.eclipse.edc:identity-hub-did-store-sql
 edc-sql-ih-keypairstore-sql = { module = "org.eclipse.edc:identity-hub-keypair-store-sql", version.ref = "edc" }
 edc-sql-ih-pcstore-sql = { module = "org.eclipse.edc:identity-hub-participantcontext-store-sql", version.ref = "edc" }
 
-# identityhub dependenciesl
+# identityhub dependencies
 edc-ih-core = { module = "org.eclipse.edc:identity-hub-core", version.ref = "edc" }
 edc-ih-keypairs = { module = "org.eclipse.edc:identity-hub-keypairs", version.ref = "edc" }
 edc-ih-did = { module = "org.eclipse.edc:identity-hub-did", version.ref = "edc" }
 edc-ih-participants = { module = "org.eclipse.edc:identity-hub-participants", version.ref = "edc" }
+edc-sts-accountprovisioner =  { module = "org.eclipse.edc:sts-account-provisioner", version.ref = "edc" }
 
 # identityhub SPI modules
 edc-ih-spi-did = { module = "org.eclipse.edc:did-spi", version.ref = "edc" }
@@ -176,7 +177,7 @@ did = ["edc-did-web", "edc-did-core", "edc-ih-did", "edc-ih-did-localpub"]
 identity-api = ["edc-ih-mgmt-config", "edc-ih-mgmt-did", "edc-ih-mgmt-participantcontext", "edc-ih-mgmt-verifiablecredential",
     "edc-ih-mgmt-keypairs", "edc-ih-mgmt-authentication", "edc-ih-mgmt-authorization"]
 
-identityhub = ["edc-ih-core", "edc-ih-api-presentation", "edc-ih-participants", "edc-ih-keypairs"]
+identityhub = ["edc-ih-core", "edc-ih-api-presentation", "edc-ih-participants", "edc-ih-keypairs", "edc-sts-accountprovisioner"]
 
 dcp = ["edc-dcp", "edc-did-core", "edc-did-web", "edc-oauth2-client", "edc-dcp-core"]
 

